.github/workflows/helm-multi-release.yaml

```diff
@@ -0,0 +1,183 @@
+        description: "The GitHub username for committing the changes."
+        required: true
+      github-email:
+        description: "The GitHub email for committing the changes."
+        required: true
+      github-token:
+        description: "The GitHub token for committing the changes."
+        required: true
+
+jobs:
+  helm-lint:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Lint Helm chart
+        uses: bakdata/ci-templates/actions/helm-lint@v1.0.0
+        with:
+          lint-config-path: "${{ inputs.lint-config-path }}"
+
+  update-version:
+    runs-on: ubuntu-22.04
+    needs: [helm-lint]
+    env:
+      ref: ${{ github.event.repository.default_branch }}
+      release-tag: ${{github.ref_name}}
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.ref }}
+
+      - name: Use the tag to update chart.yaml files
+        run: |
+          sudo add-apt-repository ppa:rmescandon/yq
+          sudo apt update
+          sudo apt install yq -y
+          for subdir in */; do
+            yq e -i '.version = "${{  env.release-tag }}"' "$subdir/Chart.yaml"
+          done
+        shell: bash
+        working-directory: ${{ inputs.charts-dir }}
+
+      - name: Commit and push updated charts
+        uses: bakdata/ci-templates/actions/commit-and-push@v1.6.0
+        with:
+          add-untracked: true
+          ref: ${{ env.ref }}
+          commit-message: "Update Chart versions"
+          github-username: ${{ secrets.github-username }}
+          github-email: ${{ secrets.github-email }}
+          github-token: ${{ secrets.github-token }}
+
+  package-new-artifacts:
+    strategy:
+      matrix:
+        subdir: ${{ fromJson(inputs.subdirs) }}
+    runs-on: ubuntu-22.04
+    needs: [update-version]
+    env:
+      charts-subdir: "${{ matrix.subdir }}"
+      artifact-dir: "${{ inputs.artifact-dir }}"
+      ref: ${{ github.event.repository.default_branch }}
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.ref }}
+
+      - name: Prepare artifacts
+        run: |
+          mkdir -p ${{ env.artifact-dir }}/${{ env.charts-subdir }}
+          helm package ./${{ inputs.charts-dir }}/${{ env.charts-subdir }} -d ${{ env.artifact-dir }}/${{ env.charts-subdir }}
+        shell: bash
+        working-directory: ${{ inputs.project-root-dir }}
+      
+      - name: Upload artifacts with action/upload
+        uses: actions/upload-artifact@v3.1.1
+        with:
+          name: ${{ env.charts-subdir }}
+          path: ${{ env.artifact-dir }}/${{ env.charts-subdir }}
+          retention-days: 1
+
+  publish-artifacts:
+    runs-on: ubuntu-22.04
+    needs: [package-new-artifacts]
+    env:
+      gh-pages: ${{ inputs.gh-pages-branch }}
+      root-dir: ${{ inputs.project-root-dir }}
+      artifact-dir: "${{ inputs.artifact-dir }}"
+      charts-dir: "${{ inputs.artifact-dir }}"
```

Comment:
If the env vars are not specifically needed please just use the inputs directly as it reduces complexity a bit